pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4655,6 +4655,68 @@ protected void internalGetLastMessageId(AsyncResponse asyncResponse, boolean aut
                     return null;
         });
     }
+    protected CompletableFuture<Void> internalTrimTopic(AsyncResponse asyncResponse, boolean authoritative) {
+        return validateTopicOwnershipAsync(topicName, authoritative)
```

Comment:
We should get the partitions first? If it's a partitioned topic, it doesn't make sense to check the ownership of the partitioned topic.